ducktape/services/service.py

```diff
@@ -108,13 +106,13 @@ def setup_node_spec(num_nodes=None, node_spec=None):
         else:
             try:
                 for os_type in RemoteAccount.SUPPORTED_OS_TYPES:
-                    if not node_spec[os_type]:
+                    if os_type not in node_spec:
```

Comment:
We don't have to. We can assume 0. I like that -- it's more flexible. I'll make that change.